tkgbot.py

```diff
@@ -169,40 +169,87 @@ def __call__(self, chat_id, args):
 class ShowCommand(BotCommand):
     help = _('''Show your subscriptions''')
 
-    def __call__(self, chat_id, args):
-        subs = self.bot.session.query(Subscription).filter_by(chat_id=chat_id).all()
+    def _get_node_description(self, node_id):
+        node_descriptions = {
+            NodeType.ALL: _('Whole forum'),
+            NodeType.MATERIAL: _('Materials'),
+            NodeType.EVENT: _('Events'),
+            NodeType.TOPIC: _('Topics'),
+            NodeType.NEWS: _('News'),
+        }
+
+        try:
+            result = node_descriptions.get(NodeType(node_id), '')
+        except ValueError:
+            result = ''
 
-        if not subs:
-            return _('You have no subscriptions')
+        return result
 
-        msg_list = []
+    def _get_node_str(self, node_id):
         node_strs = {
             NodeType.ALL: 'all',
             NodeType.MATERIAL: 'materials',
             NodeType.EVENT: 'events',
             NodeType.TOPIC: 'topics',
             NodeType.NEWS: 'news',
         }
+        
+        try:
+            result = node_strs.get(NodeType(node_id), '')
+        except ValueError:
+            result = ''
+
+        return result
+
+    def _build_subscription_info(self, sub):
+        line = ['  -'] # What are you looking at?
+
+        node_description = self._get_node_description(sub.node_id)
+        if node_description:
+            line.append(f' {node_description}')
+        elif sub.node.name and sub.node.name:
+            line.append(f' "{sub.node.name}"')
+
+        if not sub.exception:
+            if sub.no_replies or sub.no_comments:
```

Comment:
exception, no-replies and no-comments are already mutually excluding. This check is done upon submitting a subscription in such a way, that only one of these flags can be set at a time, so maybe there is not need to check it here.
See the `SubscribeCommand`.

Reply:
My suggestion is that I can't rely on things that are not reflected in code (or data structures).
So yes, technically this check is redundant but it makes code more resistant to possible future errors.

As an option - something like
`# TODO make subscription type as enum`
can be added to describe this line.

Reply:
This is a valid point.